src/EventListener/SessionListener.php

```diff
@@ -77,7 +77,7 @@ public function onKernelRequest(SessionRequestEvent $event)
 
     public function onKernelResponse(SessionResponseEvent $event)
     {
-        if (!$event->isMasterRequest()) {
+        if (Kernel::MAJOR_VERSION >= 6 ? !$event->isMainRequest() : !$event->isMasterRequest()) {
```

Comment:
I would suggest doing a method_exists check, this to avoid triggering deprecations on symfony 5.4

```
$mainRequest = method_exists($event, 'isMainRequest') ? $event->isMainRequest() : $event->isMasterRequest();
```

Reply:
ah good point. we could also version compare >= 5.3

it seems cleaner to me to check the kernel version than doing the method_exists check. do you think that would make sense?

Reply:
In this case it wouldn't indeed make a difference as the event class used here is in the same package as the Kernel class, but most of the time when I need those checks I use the method_exists for 2 reasons.

* The kernel version might be different as the package version of that specific package (if it's not located in the http-kernel package ofcourse)
* Calling the deprecated method when the new one already exists (in this case 5.4 has the method also) will still trigger deprecations on 5.4 which will make the user think that he needs to resolve it still somehow.